Automattic/jetpack
# AI Logo Generator: improve extension availability check

Fixes #38600.

## Proposed changes:

* Add a check for the `AI Assistant` block availability, as a proxy for Jetpack AI being enabled or not
* Add a check for the `AI Assistant` block visibility on the editor, as a proxy for the will of the user to hide AI features on the editor

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor, add a site logo block and check the toolbar for the AI extension button
* The AI button should be visible when:
   * The beta blocks are enabled &&
   * The AI Assistant block is registered &&
   * The AI Assistant block is visibile on the editor
* Experiment changing some of the variables and confirm the AI extension is gone:
   * Change blocks variation from beta to production
   * Disable the AI Assistant block using a snippet with `add_filter( 'jetpack_ai_enabled', '__return_false' )`
   * Hide the AI Assistant block on the block editor: go to the Gear icon on the top right corner > Preferences > Blocks > Uncheck the AI Assistant block > reload the editor
